src/3p-frame.js

```diff
@@ -212,22 +212,30 @@ export function resetBootstrapBaseUrlForTesting(win) {
  * @return {string}
  */
 export function getDefaultBootstrapBaseUrl(parentWindow, opt_srcFileBasename) {
-  if (parentWindow.defaultBootstrapBaseUrl) {
-    return parentWindow.defaultBootstrapBaseUrl;
-  }
   const srcFileBasename = opt_srcFileBasename || 'frame';
+  // Cached at filename level to ensure different url is returned.
+  parentWindow.defaultBootstrapBaseUrl =
+      parentWindow.defaultBootstrapBaseUrl || {};
+  if (parentWindow.defaultBootstrapBaseUrl[srcFileBasename]) {
+    return parentWindow.defaultBootstrapBaseUrl[srcFileBasename];
+  }
   if (getMode().localDev || getMode().test) {
     if (overrideBootstrapBaseUrl) {
-      return parentWindow.defaultBootstrapBaseUrl = overrideBootstrapBaseUrl;
+      return parentWindow.defaultBootstrapBaseUrl[srcFileBasename] =
```

Comment:
Just want to point out that this currently only allows a single override for all opt_srcFileBasename's. I'm assuming that's fine for now and we can cross the bridge if the tests ever necessitate.

Reply:
Right, I wanted to make minimal changes required.